### PR TITLE
Serialize requests to json

### PR DESCRIPTION
**Pull request by Tamir-Schwarz**
link to issue
https://github.com/opensearch-project/opensearch-java/issues/588

actually used the code of @dblock  - thank you very much!

original example
https://github.com/dblock/opensearch-java-client-demo/blob/to-json/src/main/java/Example.java#L34

=========
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

File: java-client/src/test/java/org/opensearch/client/opensearch/json/JsonDataTest.java [hunk 1]
```diff
@@ -89,4 +96,65 @@ public void testConvert() {
...
+        String expectedJson = "{\"aga\":42,\"name\":\"koko\"}";
+        Person person = new Person(42, "koko");
+        IndexRequest<Person> indexRequest = new IndexRequest.Builder<Person>()
+                .index("sample-index")
+                .id("1")
+                .document(person)
+                .build();
+
+        String json = JsonpUtils.toJson(indexRequest);
+        System.out.println(json);
```
Comment (dblock) [inline java-client/src/test/java/org/opensearch/client/opensearch/json/JsonDataTest.java:132]: Get rid of print outs.

**Review by dblock — CHANGES_REQUESTED** [review]
I think with some documentation in USER_GUIDE or related this could be good enough.

**Comment by VachaShah** [comment]
> I think with some documentation in USER_GUIDE or related this could be good enough.

+1. Some documentation on how to use `toJson()` for a class would be great!

**Comment by dblock** [comment]
@Tamir-Schwarz Want to finish this?

**Comment by felipetoffolo-toast** [comment]
I don't think that this work. I can't find toJson function

**Comment by Jai2305** [comment]
Hi @Tamir-Schwarz and @dblock , first of thanks for the contribution, all services of my organization are migrating from the High Level Rest Client to Java Client and very much like these kind of enhancements to be implemented, as they provide a clean code convenience to new clients.
Although I have some suggestions and I would like your opinion on them - 
* Can we not implement this method in JsonpSerializable interface as a default method, I think it will be a better design as it will ensure that only classes implementing the interface are allowed to utilize the default method, you could internally call serialize method of the interface, this will avoid implementation and call to a utils class.
* Also you can try implementing `JsonGenerator generator = mapper.jsonProvider().createGenerator(stringWriter);` with try with resource block to ensure that  the generator is closed automatically even if an exception occurs.
* Finally please include test cases for classes which are implementing JsonpSerializable indirectly, just to ensure the robustness of code , for eg. IndexResponse extends WriteResponseBase implements JsonpSerializable .

**Comment by dblock** [comment]
@Jai2305 These look very reasonable. Want to open another PR with your suggestions?

**Comment by Jai2305** [comment]
@dblock thanks for a quick response, I am on it.

**Comment by Xtansia** [comment]
A version of this utility was added via #588, documented here: https://github.com/opensearch-project/opensearch-java/blob/main/guides/json.md#using-tojsonstring

**Pull request closed without merging**